### ISSUE TITLE
Adapt to Taskcluster Checks API integration

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#14846]
In https://github.com/web-platform-tests/wpt.fyi/issues/1020 @lukebjerring and I discovered that with the Checks API no new check suite is created when two branches are pushed with the same sha. That means that an epochs/daily branch trailing master doesn't work. The same problem will probably affect Taskcluster.

@jgraham do you have a contact for the Checks API work for Taskcluster? Will our use of epochs/* branches break?

**Comment by jgraham** [comment]
@owlishDeveloper implmenented the support so might be able to help. Looking at the tc repo it seems like the change to enable it at all is very straightforward: https://github.com/taskcluster/taskcluster/commit/14b731786be8f4befeeea136a238bf5dea3e2e06

**Comment by owlishDeveloper** [comment]
We were discussing this on IRC. I am not quite imagining/understanding your use case, therefore cannot tell with certainty whether your use of use of epochs/* branches will break. You can either try (you can always go back) or I could test it for you, but then I need to understand how to reproduce your situation.

All I can say, we don't do anything about check suites. GitHub creates them automatically, and we just populate them with check runs. Nothing has changed in terms of how Taskcluster reacts to Github events - that part stays the same.

**Comment by foolip** [comment]
https://github.community/t5/GitHub-API-Development-and/Can-the-Checks-API-be-used-to-trigger-two-runs-when-two-pushed/m-p/18175/highlight/true#M758 shows what GitHub Actions does in this case. Having Taskcluster do the same and create two check suites would be great. I believe that will require manually creating checks suites, though.

**Comment by stephenmcgruer** [comment]
Forgot to assign this to myself. We started working on this a few weeks ago; I have an [overview document](https://docs.google.com/document/d/1rMgrxewvywcFQ2_vc0bBWIP1kykWJlHSfZRn6D_ww1M/edit#) that tracks the work.

At this point, wpt.fyi (and staging.wpt.fyi) should be able to receive `check_suite` events via the normal webhook, and parse them similarly to the `status` events used today. This [has been tested](https://docs.google.com/document/d/1rMgrxewvywcFQ2_vc0bBWIP1kykWJlHSfZRn6D_ww1M/edit#heading=h.q8tyq0wwfbo5) via a clone of WPT, [wpt-tc-checks](https://github.com/web-platform-tests/wpt-tc-checks). The roll-out plan would look like:

- [ ] Enable `check_suite` events for the wpt.fyi and staging.wpt.fyi webhooks in the main wpt repo. This can happen anytime; the endpoint will ignore non-Taskcluster events, and Taskcluster won't cause check_suite events until...
- [ ] Cherry-pick the added support in the wpt-tc-checks repository to make Taskcluster use their checks integration rather than their status one.
- [ ] Update the set of required checks in WPT to include the new 'sink' task that was added, and remove the old 'Community-TC' check.
- [ ] Monitor new PRs to make sure the expected things are happening.
- [ ] Deal with existing PRs by close/opening them, which should cause the new checks to run instead.

If we find an issue along the way, we need to revert the cherry-pick, and change the required checks back.

**Comment by stephenmcgruer** [comment]
>  Cherry-pick the added support in the wpt-tc-checks repository to make Taskcluster use their checks integration rather than their status one.

This is now ready; https://github.com/web-platform-tests/wpt/pull/24151

My plan currently is to attempt the migration on the (EST) morning of Tuesday, July 7th. I will send an announcement to the reviewers team on web-platform-tests, as well as on IRC, before I do so. After flipping the required bit, I will be going through active PRs and re-triggering their checks as necessary.

**Comment by stephenmcgruer** [comment]
The PR has landed, and the expected checks have been updated.

We have hit one snag - I overlooked the fact that `.taskcluster.yml` has changed so PRs need **rebased** rather than close/opened to retrigger the CI. This is obviously a much more painful bar, so we're considering our options for folks atm.

**Comment by stephenmcgruer** [comment]
Update: so far things are looking good on the GitHub side, and pull request integration with wpt.fyi appears to be working, but for some reason full pushes to wpt.fyi are missing data. https://wpt.fyi/results/?diff&filter=ADC&run_id=593330004&run_id=593410002 is a diff from a run yesterday to today; we appear to be dropping shards.

https://github.com/web-platform-tests/wpt.fyi/pull/2044 has added more logging and we're debugging the problem.

**Comment by stephenmcgruer** [comment]
We believe we have identified the problem and have a fix: https://github.com/web-platform-tests/wpt.fyi/pull/2045

**Comment by stephenmcgruer** [comment]
The latest runs look good; we're now looking at a problem with manifest actions but we think it may be unrelated.

**Comment by stephenmcgruer** [comment]
The manifest problem turned out to be unrelated (and has been corrected); we also discovered an unrelated problem with the epochs generation (https://github.com/web-platform-tests/wpt/issues/24504). 

There are two outstanding problems:
  * The sink-task can require >100 dependencies for daily epoch runs, which is higher than what Taskcluster supports. The office workaround is to shard your sink-task, but instead we have determined we don't need the sink-task for anything other than pull requests anyway (https://github.com/web-platform-tests/wpt/pull/24511)
  * The workaround for pagination on wpt.fyi (https://github.com/web-platform-tests/wpt.fyi/pull/2045) is just that, a workaround, and we will need to put a proper fix in place given that things like daily epoch runs have 100s of tasks.
